src/darwin/Framework/CHIP/MTRBaseDevice.h

```diff
@@ -605,22 +648,62 @@ typedef NS_ENUM(NSUInteger, MTREventPriority) {
 
 @interface MTREventReport : NSObject
 @property (nonatomic, readonly, copy) MTREventPath * path;
+
+/**
+ * eventNumber will only have a useful value if "error" is nil.
+ */
 @property (nonatomic, readonly, copy) NSNumber * eventNumber; // EventNumber type (uint64_t)
+
+/**
+ * priority will only have a useful value if "error" is nil.
+ */
 @property (nonatomic, readonly, copy) NSNumber * priority; // PriorityLevel type (MTREventPriority)
```

Comment:
Is there a reason for using an NSNumber here when we already have a type for MTREventPriority?

Reply:
There were some API decisions early on around having everything numeric be NSNumber...  And at this point we shipped this a while back, so changing it is a bit complicated.  Note that I am just adding comments here, not changing the actual definition.

Reply:
@jake-apple ^

Reply:
> Will the compiler let you just use MTREventPriority here instead of NSNumber, and if so, should we do that?

That would be a source-and-binary-incompatible change from what we have now.